docs/index.bs

```diff
@@ -3707,6 +3986,28 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
                   1. Return true.
       1. Return false.
   </section>
+
+  <section algorithm>
+    <h3 id="lookup-race-response-algorithm"><dfn export>Lookup Race Response</dfn></h3>
+
+      : Input
+      :: |request|, a [=/request=]
+      : Output
+      :: a [=/response=] or null
+
+      1. If |request|'s [=request/reserved client=] is null, return null.
```

Comment:
nit: maybe add a Note explaining that this check ensures that we're only dealing with navigation requests, and thus looking up a registration using the request URL is correct.

Reply:
Ah, I was wrong. We don't want to limit this algorithm for non-subresource requests. What we'd like to achieve here is to get [=active worker=] from [=/request=], regardless of non-subresource requests or subresource requests.

Updated to get the active worker from subreosurce requests, but perhaps we can just use |request|'s [=request/client=]'s [=active service worker=] for both subresource and non-subresource requests?

Reply:
What you have now looks reasonable to me. I think using request's client's active service worker for non-subresource requests would be incorrect. Maybe at some point we should abstract out a "give me the right registration for this request" algorithm to be shared between Handle Fetch and this algorithm, but for now what you have seems fine too.

Reply:
Thank you for the clarification!